### PR TITLE
Change the nutrient format

### LINKED ISSUE #78: The concentration of ingredients in the form should present the value and its unit
Ingredients/nutrients concentration should be given in an object:
```json
{
	"nutrient": "Something",
	"value": "44",
	"unit": "%"
}
```

 Tasks:

 * [x] Update the prompt to return the concentration in this object.
 * [x] Create a nutrient object and use it in the form.

 Acceptance Criteria:

- All the tests should pass.
- The router must return a valid JSON.

### PR DESCRIPTION
**Pull request by snakedye**
Close #78

### CODE REVIEW

File: tests/test_form.py [hunk 1]
```diff
@@ -4,106 +4,53 @@
...
+            "specifications_en": [{"humidity": "Low", "ph": "7", "solubility": "High"}],
+            "guaranteed_analysis": [{"nutrient": "Nitrogen", "value": "10", "unit": "%"}]
+        }
+
+        try:
+            form = FertiliserForm(**data)
+        except ValidationError as e:
+            self.fail(f"Validation error: {e}")
+
+        self.assertEqual(form.company_name, "ABC Company")
```
Comment (Francois-Werbrouck) [inline tests/test_form.py:24]: Take a look at https://github.com/ai-cfia/nachet-datastore/blob/9278eda4a03d5256dcbd3d64646c44dbdca75b6e/tests/test_metadata.py#L63 This should make your testing less redundant 😉

File: tests/test_form.py [hunk 2]
```diff
@@ -4,106 +4,53 @@
...
+            "company_name": "ABC Company",
+            "fertiliser_name": "Super Fertiliser",
+            "npk": "invalid-format",  # invalid npk format
+            "instructions_en": ["Use as directed"],
+            "micronutrients_en": [{"nutrient": "Iron", "value": "2", "unit": "%"}],
+            "specifications_en": [{"humidity": "Low", "ph": "7", "solubility": "High"}],
+            "guaranteed_analysis": [{"nutrient": "Nitrogen", "value": "10", "unit": "%"}]
+        }
+
+        with self.assertRaises(ValidationError):
```
Comment (Francois-Werbrouck) [inline tests/test_form.py:52]: Shouldn't this be a module defined type of error instead of a generic ValidationError?

Comment (Francois-Werbrouck) [inline tests/test_form.py:52]: Shouldn't this be a module defined type of error instead of a generic ValidationError?

**Review by Francois-Werbrouck — CHANGES_REQUESTED** [review]

File: backend/form.py [hunk 3]
```diff
@@ -1,11 +1,12 @@
+import re
 from typing import List, Optional
 from pydantic import BaseModel, Field, field_validator
```
Comment (Francois-Werbrouck) [inline backend/form.py:4]:
```python
class npkError(ValueError):
pass
```

File: backend/form.py [hunk 4]
```diff
@@ -63,6 +64,14 @@ def convert_values(cls, v):
         if isinstance(v, (int, float)):
             return str(v)
         return v
+    
+    @field_validator('npk', mode='before')
+    def validate_npk(cls, v):
+        if v is not None:
+            pattern = re.compile(r'^\d+-\d+-\d+$')
+            if not pattern.match(v):
+                raise ValueError('npk must be in the format "number-number-number"')
```
Comment (Francois-Werbrouck) [inline backend/form.py:73]:
```python
raise npkError('npk must be in the format "number-number-number"')
```

**Review by Francois-Werbrouck — APPROVED** [review]
Having error built in your modules will help future and other devs to maintain the tool (it could be an issue of its own). It also is a way to specifically test for this error type in your test suite. Other than that LGTM!

**Review by Youllou — APPROVED** [review]

**Pull request merged**